### PR TITLE
Post a new DNS entry to the DNS API

### PR DESCRIPTION
**Pull request by samgiles**
@fabricedesre  r?

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -251,8 +254,8 @@ fn main() {
     }
 
     // Register with the nUPNP server.
-    let registrar = registration::Registrar::new();
-    registrar.start(args.flag_register, args.flag_iface, domain, &tunnel, controller.get_certificate_manager());
+    let registrar = registration::Registrar::new(controller.get_hostname());
+    registrar.start(args.flag_register, args.flag_iface, domain, &tunnel, args.flag_port, args.flag_dns_api, controller.get_certificate_manager());
```
Comment (fabricedesre) [inline src/main.rs]: nit: this line is getting too long

File: src/registration.rs [hunk 2]
```diff
@@ -73,6 +76,7 @@ impl Registrar {
             panic!("Could not get or generate self signed certificate - registration will always fail: {}", certificate_record_result.err().unwrap().description());
         };
 
+
```
Comment (fabricedesre) [inline src/registration.rs]: nit: leftover

File: src/tls/certificate_manager.rs [hunk 3]
```diff
@@ -227,18 +227,18 @@ impl CertificateManager {
...
 
-    #[allow(dead_code)]
-    pub fn register_for_dns_challenge(&mut self, hostname: String, dns_endpoint: String, ip_address: String) -> Result<bool, String> {
-        if let Some((https_client, cert_record)) = self.create_https_client_with_crt_for(hostname.clone()) {
-
-            let hash = cert_record.get_certificate_fingerprint();
+    pub fn register_dns_record(&self,
+                               record_type: String, hostname: String,
+                               ip_address: String, dns_endpoint: String,
+                               certificate: String) -> Result<bool, String> {
```
Comment (fabricedesre) [inline src/tls/certificate_manager.rs]: move all the String parameters to &str

Comment (samgiles) [inline src/tls/certificate_manager.rs]: Yeh, I need to do this across most of the APIs in `tls/`

**Comment by fabricedesre** [comment]
r=me with comments addressed

**Pull request merged**